lib/src/mocks/http.dart

```diff
@@ -36,8 +37,27 @@ void cancelMockRequest(MockBaseRequest request) {
 }
 
 handleMockRequest(MockBaseRequest request) {
-  Iterable matchingExpectations = _expectations
-      .where((e) => e.method == request.method && e.uri == request.uri);
+  Iterable matchingExpectations = _expectations.where((e) {
+    bool methodMatches = e.method == request.method;
+    bool uriMatches = e.uri == request.uri;
+    bool headersMatch;
+    if (e.headers == null) {
+      // Ignore headers check if expectation didn't specify.
+      headersMatch = true;
+    } else if (e.headers.isEmpty) {
+      headersMatch = request.headers.isEmpty;
+    } else {
+      headersMatch = true;
+      e.headers.forEach((header, value) {
```

Comment:
technically, if `request` includes headers that aren't included in `e`, this will return as a successful match, even though they are not actually a match.  Does that matter for the usage here?